azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -1728,6 +1729,28 @@ private void ajaxHandleUpload(final HttpServletRequest req, final HttpServletRes
           this.scheduler.schedule(project, user.getUserId());
         }
 
+        // remove schedule of renamed/deleted flows
+        final Set<String> flowNameList = new HashSet<>();
+        for (final Flow flow : project.getFlows()) {
+          flowNameList.add(flow.getId());
+        }
+
+        try {
+          for (final Schedule schedule : this.scheduleManager.getSchedules()) {
```

Comment:
In some clusters, we have thousands of schedules. Will this check impact the uploading time for users?

Reply:
Maybe we can measure it and test it out on a dev cluster? If this comes out to be a real problem, then we can maintain a set of scheduleIds in Project class. Then we only look up schedules for the given project.
ScheduleManager already maintain a map of scheduleID to Schedule which can be used to remove the schedule.

Reply:
I don't think the check will add considerable latency. I tested locally with a List of 1 million objects containing an Integer and the time it took to iterate and compare values is under 15 milliseconds.